sdk/data/aztables/client.go

```diff
@@ -128,6 +128,8 @@ func (t *Client) CreateTable(ctx context.Context, options *CreateTableOptions) (
 
 // Delete deletes the table with the tableName specified when NewClient was called. If the service returns a non-successful HTTP status
 // code, the function returns an *azcore.ResponseError type. Specify nil for options if you want to use the default options.
+// NOTE: deleting a table can take up to 40 seconds or more to complete.  If a table with the same name is created while the delete is still
+// in progress, an *azcore.ResponseError is returned with error code TableBeingDeleted and status code http.StatusConflict.
```

Comment:
Unfortunately, the only operation that returns a 409 is creating the table.  Other operations return a 404 so you can't spin on them to deterministically tell when the table has been deleted.  It means that callers of `CreateTable` would need to be resilient to this 409 case.

Reply:
Added doc comment to `CreateTable` explaining this.